airbyte-integrations/connectors/source-snapchat-marketing/source_snapchat_marketing/source.py

```diff
@@ -167,39 +199,53 @@ def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapp
         So the response_root_name will be "organizations", and the response_item_name will be "organization"
         Also, the client side filtering for incremental sync is used
         """
+
         json_response = response.json().get(self.response_root_name)
         for resp in json_response:
             if self.response_item_name not in resp:
-                error_text = f"JSON field named '{self.response_item_name}' is absent in the response for {self.name} stream"
+                error_text = f"stream {self.name}: field named '{self.response_item_name}' is absent in the response: {resp}"
                 self.logger.error(error_text)
                 raise SnapchatMarketingException(error_text)
             yield resp.get(self.response_item_name)
 
 
 class IncrementalSnapchatMarketingStream(SnapchatMarketingStream, ABC):
     cursor_field = "updated_at"
-    depends_on_stream = None
-    slice_key_name = "ad_account_id"
 
     last_slice = None
     current_slice = None
     first_run = True
+    initial_state = None
+    max_state = None
+
+    @property
+    @abstractmethod
+    def parent(self) -> SnapchatMarketingStream:
+        """Stream Class to extract entity ids from"""
+
+    def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
+        return f"adaccounts/{stream_slice['id']}/{self.response_root_name}"
 
     def stream_slices(self, **kwargs) -> Iterable[Optional[Mapping[str, Any]]]:
-        depends_on_stream_config = {"authenticator": self.authenticator, "start_date": self.start_date}
-        stream_slices = get_depend_on_ids(self.depends_on_stream, depends_on_stream_config, self.slice_key_name)
 
-        if not stream_slices:
-            self.logger.error(f"No {self.slice_key_name}s found. {self.name} cannot be extracted without {self.slice_key_name}.")
-            yield from []
+        stream_state = kwargs.get("stream_state")
+        self.initial_state = stream_state.get(self.cursor_field) if stream_state else self.start_date
+        self.max_state = self.initial_state
+
+        parent_stream = self.parent(authenticator=self.authenticator, start_date=self.start_date, end_date=self.end_date)
+        stream_slices = get_parent_ids(parent_stream)
+
+        if stream_slices:
+            self.last_slice = stream_slices[-1]
 
-        self.last_slice = stream_slices[-1]
-        yield from stream_slices
+        self.logger.info(f"{self.name}: stream_slices:{stream_slices}")
```

Comment:
do we want to log this in production? If so, I think the message should be more formatted because this is consumer facing.
Maybe something like 
```suggestion
        self.logger.info(f"stream slicers for {self.name}: {stream_slices}")
```

Alternatively, we can log this at the debug level so it's only visible when running with the `--debug` flag